docs/code/component-architecture/README.md

```diff
@@ -1,9 +1,132 @@
+  :::
+
+## File Structure
+
+**Every component gets its own directory**. Components should not hang out on their own as individual files. This follows the single-responsibility principle, and is built upon [this practice](https://cobwwweb.com/organize-components-by-keeping-related-files-close) Sean laid out in keeping related files close to make it easier to move around a project's codebase.
+
+Within that directory, there may be some variation, but the structure should largely look the same across all components. A typical component (if there is such a thing) looks like this:
+
+```
+[component]/
+├── adapter.js
+├── component.js
+├── fixtures.js
+├── index.js
+├── styles.module.scss
+└── test.spec.js
+```
+
+See below for explanations on each piece of the puzzle:
+
+### `adapter.js`
+
+Of all the files shown above, the adapter is likely the one you'll see the least frequently. Its role is to feed in dynamic data into a component. It connects directly to the data source, pulls in the data, transforms it, and then sends it on to the component.
+
+You can learn more about the theory behind this approach in [this article](https://cobwwweb.com/simplify-components-by-separating-logic-from-presentation-using-adapters.html). Sean also wrote [an article on a Gatsby-specific implementation](https://cobwwweb.com/introducing-component-adapters-into-gatsby.html) that may be worthwhile if you want to learn more about the approach.
+
+### `component.js`
+
+This is the main component. If you're used to only ever seeing a single file represent a component, this is that file.
+
+### `fixtures.js`
+
+Fixtures are static sets of data that primarily serve two purposes:
+
+1. They feed into components to provide prop values during development. This enables front-end developers to move quickly without worrying about the state or structure of the data source.
+2. They are used for testing (this is the traditional use of a [fixture](https://en.wikipedia.org/wiki/Test_fixture)).
+
+### `index.js`
+
+The index file is the entry point for the files within the project. Often it looks something like this:
+
+```jsx
+import component from "./component"
+import fixtures from "./fixtures"
+
+export default component
+
+export { component, fixtures }
+```
+
+What this does is provide a front door for the component, while enabling other important pieces to be included. This provides us with three primary benefits:
+
+1. When a component is consumed by another, it doesn't matter what the structure is — the component directory's default export is automatically picked up, as long as the consuming component imports the component from the index file.
+2. It puts all relevant exports in one place. Consider that a test spec could load both the component and the fixture from a single file.
+3. It provides a means to contextually switch the default export depending on the environment. This comes in handy when working with adapters.
```

Comment:
Would you possibly have an example in the wild of this used with an adapter? I think it may be helpful to understand what it's doing since adapters are a little more (somewhat) complex topic.

It took me a minute to understand what the `export { component, fixtures }` was used for since I can `import fixtures from "component/path/fixtures"` the same as  `import { fixtures } from "component/path"`

I eventually saw value in it since you can do `import { fixtures, adapter, component as MyComponent } from "component/path"` along with the simplicity of still having `import MyComponent from "component/path"`

Reply:
Great feedback — thank you. I added a living example in the `adapter.js` section, and a tip here about the benefit of named exports.